lib/market.rb

```diff
@@ -1,3 +1,42 @@
 class Market
- 
+    attr_reader :name, :vendors
+
+    def initialize(name)
+        @name = name
+        @vendors = ([])
+    end
+
+    def add_vendor(vendor)
+        @vendors << vendor
+    end
+
+    def vendor_names
+        @vendors.map { |vendor| vendor.name }
+    end
+
+    def vendors_that_sell(item)
+        @vendors.select do |vendor|
+            vendor.inventory.include?(item)
+        end
+    end
+
+    def sorted_item_list
+        vendors.flat_map { |vendor| vendor.inventory.keys.map(&:name)}.uniq.sort
+    end
+
+    def total_inventory
+        all_inventory = {}
+
+        @vendors.each do |vendor|
```

Comment:
Theres probably an easier way to do this. Ask perplexity for some refactoring tips!